pkg/paths/path.go

```diff
@@ -0,0 +1,155 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package paths
+
+import (
+	"strings"
+)
+
+//
+// To avoid confusion about when paths are encrypted, unencrypted, empty or
+// non existent, we create some wrapper types so that the compiler will complain
+// if someone attempts to use one in the wrong context.
+//
+
+// Unencrypted is an opaque type representing an unencrypted path.
+type Unencrypted struct {
+	raw string
```

Comment:
We need to represent these somehow better since the encrypted path segment could can contain `/`, and then this wouldn't properly escape.

Reply:
Also, why not using the `storj.Path` alias instead of `string`?

Reply:
This would never contain an encrypted path segment. It's an Unencrypted path. It's not using the same alias because I would like that alias to be removed so that we're statically clear about what all the `storj.Path` types contain (is it encrypted? unencrypted? contain a bucket? contain a segment? contain a project id? etc, etc, etc.)